rest_api/db/accounts_query.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import rethinkdb as r
+from rethinkdb.errors import ReqlNonExistenceError
+
+from api.errors import ApiBadRequest
+
+from db import auth_query
+from db import blocks_query
+
+
+async def fetch_all_account_resources(conn):
+    return await r.table('accounts')\
+        .filter((blocks_query.latest_block_num() >= r.row['start_block_num'])
+                & (blocks_query.latest_block_num() < r.row['end_block_num']))\
+        .map(lambda account: account.merge(
+            {
+                'publicKey': account['public_key'],
+                'email': auth_query.fetch_email(account['public_key'])
+            }))\
+        .map(lambda account: (account['label'] == "").branch(
+            account.without('label'), account))\
+        .map(lambda account: (account['description'] == "").branch(
+            account.without('description'), account))\
+        .without('public_key', 'delta_id',
+                 'start_block_num', 'end_block_num')\
+        .coerce_to('array').run(conn)
+
+
+async def fetch_account_resource(conn, key):
+    try:
+        return await r.table('accounts')\
+            .get_all(key, index='public_key')\
+            .max('start_block_num')\
+            .merge(
+                {
+                    'publicKey': r.row['public_key'],
+                    'email': auth_query.fetch_email(key)
```

Comment:
Auth info should only get sent back with `GET /accounts/{key}` if the user is authorized and fetching info for themselves.